flink-connectors/flink-connector-aws-kinesis-data-streams/src/test/java/org/apache/flink/connector/kinesis/sink/KinesisDataStreamsSinkITCase.java

```diff
@@ -0,0 +1,347 @@
+        private int sizeOfMessageBytes = 25;
+        private int bufferMaxTimeMS = 1000;
+        private int maxInflightReqs = 1;
+        private int maxBatchSize = 50;
+        private int expectedElements = 50;
+        private boolean failOnError = false;
+        private String kinesaliteStreamName;
+        private String sinkConnectionStreamName;
+        private ElementConverter<String, PutRecordsRequestEntry> elementConverter =
+                KinesisDataStreamsSinkITCase.this.elementConverter;
+
+        public void runScenario() throws Exception {
+            prepareStream(kinesaliteStreamName);
+
+            DataStream<String> stream =
+                    env.addSource(
+                                    new DataGeneratorSource<String>(
+                                            RandomGenerator.stringGenerator(sizeOfMessageBytes),
+                                            100,
+                                            (long) numberOfElementsToSend))
+                            .returns(String.class);
+
+            Properties prop = new Properties();
+            prop.setProperty(AWS_ENDPOINT, KINESALITE.getHostEndpointUrl());
+            prop.setProperty(AWS_ACCESS_KEY_ID, KINESALITE.getAccessKey());
+            prop.setProperty(AWS_SECRET_ACCESS_KEY, KINESALITE.getSecretKey());
+            prop.setProperty(AWS_REGION, KINESALITE.getRegion().toString());
+            prop.setProperty(TRUST_ALL_CERTIFICATES, "true");
+            prop.setProperty(HTTP_PROTOCOL_VERSION, "HTTP1_1");
+
+            KinesisDataStreamsSink<String> kdsSink =
+                    KinesisDataStreamsSink.<String>builder()
+                            .setElementConverter(elementConverter)
+                            .setMaxTimeInBufferMS(bufferMaxTimeMS)
+                            .setMaxInFlightRequests(maxInflightReqs)
+                            .setMaxBatchSize(maxBatchSize)
+                            .setFailOnError(failOnError)
+                            .setMaxBufferedRequests(1000)
+                            .setStreamName(sinkConnectionStreamName)
+                            .setKinesisClientProperties(prop)
+                            .setFailOnError(true)
+                            .build();
+
+            stream.sinkTo(kdsSink);
+
+            env.execute("KDS Async Sink Example Program");
+
+            String shardIterator =
+                    kinesisClient
+                            .getShardIterator(
+                                    GetShardIteratorRequest.builder()
+                                            .shardId(DEFAULT_FIRST_SHARD_NAME)
+                                            .shardIteratorType(ShardIteratorType.TRIM_HORIZON)
+                                            .streamName(kinesaliteStreamName)
+                                            .build())
+                            .get()
+                            .shardIterator();
+
+            assertEquals(
```

Comment:
[We should now be using](https://flink.apache.org/contributing/code-style-and-quality-common.html#testing) `AssertJ`, please migrate tests, example:
```
Assertions.assertThat(kinesisClient
                            .getRecords(GetRecordsRequest.builder().shardIterator(shardIterator).build())
                            .get()
                            .records()
                            .size())
                            .isEqualTo(expectedElements);
```